test/integration/backward_compatible/topic/ReliableTopicOnClusterRestart.js

```diff
@@ -0,0 +1,194 @@
+            await assertTrueEventually(async () => {
+                if (!messageArrived)
+                    throw new Error('message not arrived yet');
+            }, undefined, 5000);
+        });
+
+        messageCount.should.be.greaterThanOrEqual(1);
+    });
+
+    it('should continue on cluster restart after invocation timeout', async function () {
+        const invocationTimeoutMillis = 2000;
+
+        client1 = await Client.newHazelcastClient({
+            clusterName: cluster.id,
+            connectionStrategy: {
+                connectionRetry: {
+                    clusterConnectTimeoutMillis: Number.MAX_SAFE_INTEGER
+                }
+            },
+            properties: {
+                'hazelcast.client.invocation.timeout.millis': invocationTimeoutMillis
+            }
+        });
+
+        let messageArrived = false;
+        const topicName = 'topic';
+
+        const topic1 = await client1.getReliableTopic(topicName);
+        topic1.addMessageListener(() => {
+            messageArrived = true;
+        });
+
+        await RC.shutdownMember(cluster.id, member.uuid);
+
+        // wait for the topic operation to timeout
+        await promiseWaitMilliseconds(invocationTimeoutMillis);
+
+        await RC.startMember(cluster.id);
+
+        client2 = await Client.newHazelcastClient({
+            clusterName: cluster.id,
+            connectionStrategy: {
+                connectionRetry: {
+                    clusterConnectTimeoutMillis: Number.MAX_SAFE_INTEGER
+                }
+            }
+        });
+
+        const topic2 = await client2.getReliableTopic(topicName);
+        await topic2.publish('message');
+        await assertTrueEventually(async () => {
+            if (!messageArrived)
+                throw new Error('message not arrived yet');
+        });
+    });
+
+    it('should continue on cluster restart', async function () {
+        client1 = await Client.newHazelcastClient({
+            clusterName: cluster.id,
+            connectionStrategy: {
+                connectionRetry: {
+                    clusterConnectTimeoutMillis: Number.MAX_SAFE_INTEGER
+                }
+            }
+        });
+        client2 = await Client.newHazelcastClient({
+            clusterName: cluster.id,
+            connectionStrategy: {
+                connectionRetry: {
+                    clusterConnectTimeoutMillis: Number.MAX_SAFE_INTEGER
+                }
+            }
+        });
+
+        const topicName = 'topic';
+        const topic1 = await client1.getReliableTopic(topicName);
+        const topic2 = await client2.getReliableTopic(topicName);
+
+        let messageArrived = false;
+
+        topic1.addMessageListener(() => {
+            messageArrived = true;
+        });
+
+        await RC.shutdownMember(cluster.id, member.uuid);
+        await RC.startMember(cluster.id);
+
+        await topic2.publish('message');
+        await assertTrueEventually(async () => {
```

Comment:
similar thing as above

Reply:
these are done in https://github.com/hazelcast/hazelcast-nodejs-client/pull/890/commits/c6f8d1483c1ace172ab81f96dab1e495382cc8ad